### PR TITLE
FINERACT-2081: Add all loan term validations to loan details

### PR DESCRIPTION
**Pull request by oleksii-novikov-onix**
## Description

Introduced a new field for the loan details: loanTermVariations which should contains all the loan term variations (no filtering by type).

Implemented the same for the LoanAccountData business external events as well.

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by adamsaghy** [comment]
@oleksii-novikov-onix Please squash your commits!

**Comment by oleksii-novikov-onix** [comment]
> @oleksii-novikov-onix Please squash your commits!

done

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/ExternalBusinessEventTest.java [hunk 1]
```diff
@@ -265,55 +265,48 @@ private void logBusinessEvents(List<ExternalEventDTO> allExternalEvents) {
         });
     }
 
-    public void verifyBusinessEvents(BusinessEvent... businessEvents) {
-        List<ExternalEventDTO> allExternalEvents = ExternalEventHelper.getAllExternalEvents(requestSpec, responseSpec);
+    public void verifyBusinessEvents(final BusinessEvent... businessEvents) {
+        final var allExternalEvents = ExternalEventHelper.getAllExternalEvents(requestSpec, responseSpec);
```
Comment (galovics) [inline integration-tests/src/test/java/org/apache/fineract/integrationtests/ExternalBusinessEventTest.java]: Not a big fan of var, what's the reason of changing these?

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientLoanIntegrationTest.java [hunk 2]
```diff
@@ -6923,6 +6927,236 @@ public void testLoanTransactionOrderAfterReverseReplay() {
         }
     }
 
+    @Test
```
Comment (galovics) [inline integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientLoanIntegrationTest.java]: These are super bad quality tests and hard to maintain (not specifically yours, but in general these type of tests). Can you please migrate it to using the BaseLoanIntegration test at least or implementing this functional test in the cucumber tests? The latter is preferable.

**Review by galovics — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/ExternalBusinessEventTest.java (hunk 1 above)
Comment (oleksii-novikov-onix) [reply integration-tests/src/test/java/org/apache/fineract/integrationtests/ExternalBusinessEventTest.java]: Using var has become a habit for me from working on another long-term project. I see only one drawback: the code review process in GitHub can get a bit complicated. I plan to follow the existing coding style of this project. So should I stop using var altogether? Are there specific cases where it’s allowed or prohibited?

Comment (galovics) [reply integration-tests/src/test/java/org/apache/fineract/integrationtests/ExternalBusinessEventTest.java]: @oleksii-novikov-onix I wouldn't say it's prohibited, it's just not appearing right now in the codebase and I'd keep it out just for consistency.

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientLoanIntegrationTest.java (hunk 2 above)
Comment (oleksii-novikov-onix) [reply integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientLoanIntegrationTest.java]: Implemented e2e tests

**Review by galovics — APPROVED** [review]

**Pull request merged**